ampd/src/starknet/verifier.rs

```diff
@@ -0,0 +1,53 @@
+use axelar_wasm_std::voting::Vote;
+use mockall::automock;
+use starknet_providers::jsonrpc::HttpTransport;
+use thiserror::Error;
+use tonic::async_trait;
+use url::Url;
+
+use super::events::contract_call::ContractCallEvent;
+use super::json_rpc::{Client, StarknetClientError};
+use crate::handlers::starknet_verify_msg::Message;
+
+#[derive(Error, Debug)]
+pub enum VerifierError {
+    #[error("JSON-RPC error")]
+    JsonRPC,
+    #[error("block number missing in JSON-RPC response for finalized block")]
+    MissBlockNumber,
+    #[error("failed to fetch event: {0}")]
+    FetchEvent(#[from] StarknetClientError),
+}
+
+#[automock]
```

Comment:
question: Do we use those mocks anywhere?

Reply:
I left it by mistake. This is fixed in the next PR.